### PR TITLE
Feature/mocha testing

### PR DESCRIPTION
**Pull request by knite51**
#### What does this PR do?
- Write Test cases using Mocha/Expect.
#### Description of Tasks to be completed?
-  Install mocha and Expect
- Create a src directory and open a tests directory in it.
- Create a test file as specified (test.spec.js).
- Import modules into the test.spec.js file.
- Write test for the modules inside the test.spec.js file.
- RUN: '$ npm run test' in your terminal to check test result.
#### How should this be manually tested?
- RUN: '$ npm run test' in your terminal.
#### Any background context you want to provide?
- Mocha/Expect helps write test cases, allowing for flexible and accurate reporting on codebase
#### What are the relevant pivotal tracker stories?
- https://www.pivotaltracker.com/story/show/153435435
#### Screenshots (if appropriate)
- n/a
#### Blockers (if any)
- n/a
#### Questions:
- n/a

### CODE REVIEW

File: filter.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
 /* eslint-disable no-console */
```
Comment (ghost) [inline filter.js]: Remove that comment. No longer necessary.

File: index.js [hunk 2]
```diff
@@ -7,17 +7,15 @@ class ShoppingCart {
     this.total += price * quantity;
     if (this.items[itemName] !== undefined) {
```
Comment (ghost) [inline index.js]: Change this line to `if (!this.items.itemName)` (No need for the `!==undefined`)

File: index.js [hunk 3]
```diff
@@ -7,17 +7,15 @@ class ShoppingCart {
...
       this.items[itemName] += quantity;
```
Comment (ghost) [inline index.js]: Change all instances of the `[]` here to dot notation.

**Review by ghost — CHANGES_REQUESTED** [review]
See the comments both here and on PT

File: src/tests/test.spec.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import expect from 'expect';
+import { arr, arrString } from '../../filter';
+import { Shop, ShoppingCart } from '../../index';
```
[the remark this answers is not in the document]
Comment (oreoluwade) [reply src/tests/test.spec.js]: To fix this, add this to the rules `"import/extensions": 0`

[the remark this answers is not in the document]
Comment (oreoluwade) [reply src/tests/test.spec.js]: Add this too. But not under rules, but as a standalone key-value pair.

`  "settings": {
    "import/extensions": [".js", ".jsx"]
  },`

File: index.js (hunk 2 above)
Comment (knite51) [reply index.js]: you sure it is not suppose to be this.items.itemName? putting the ! means it doesnt exist. And am checking if it does

Comment (ghost) [reply index.js]: Oh, you're right.

File: index.js (hunk 3 above)
Comment (knite51) [reply index.js]: i realized that not all [] can be changed to dot notation. It disrupt the test cases.

Comment (ghost) [reply index.js]: Oh okay. Change the ones that can be changed then, and leave the others.

File: .eslintrc [hunk 5]
```diff
@@ -2,6 +2,10 @@
   "extends": "airbnb-base",
   "rules": {
     "linebreak-style": 0,
-    "arrow-body-style":"off"
+    "arrow-body-style":"off",
+    "import/no-unresolved": [0]
+  },
+  "env":{
+    "mocha": true 
   }
```
Comment (ghost) [inline .eslintrc]: add this to the `.eslintrc`

`  "settings": {
    "import/extensions": [".js"]
  },`

File: index.js (hunk 3 above)
Comment (knite51) [reply index.js]: Okay

**Comment by knite51** [comment]
Merge?

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
Yes

**Pull request merged**